### DISCUSSION TITLE
windowsでremap(QMK)のTAP/HOLDが不安定

### DISCUSSION THREAD

**Discussion by aikawa9376** [Yowkees/keyball#252] [Q&A] [answered]
keyball39を先日組み終わり一通りの動作チェックを終えてキーマップを設定している状態です。
単独のチェックでは各キーが問題なく動いていますが、
TAP/HOLDを設定したところwindows10の環境でTAPが反応せず不安定な動きをしています。
firmwareはKeyball39_viaをremapで調整しています。
(事象としてはqmk_firmwareの[このissue](https://github.com/qmk/qmk_firmware/issues/16855)と同じだと思いますがケーブルの抜き差しでなおらない)

具体的には下記のような状態です
・キーを素早く一回押して離すと何もキーコードが入力されない
・キーを二回素早くおすとキーコードが入力される

linux(arch)につないで動かすと全て正常に動いています。

あまりwindows環境でトラブルがあるような事例を見つけることができなかったのでご質問した次第です。

TAP/HOLD以外を設定すれば各キーは問題なく動いているので
ハンダ関係では無いように思っているのですが、基盤のトラブルは考えられますでしょうか？
基盤の問題ではなさそうでしたら、QMK firmwareの調整を自身で行ってみようと考えています。

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。
私もWindows10環境で使用しておりますが、TAP/HOLDうまく機能しています。

ご指摘通り、TAP/HOLD以外を設定すれば各キーは問題なく動いているということで、
ソフト（QMK_Firmware）上での問題と思われます。

調査・調整ができそうでしたらお手数ですがご検討いただければ幸いです。
お力になれず申し訳ありません。

**Reply by aikawa9376** [reply]
お返事ありがとうございます。
問題の切り分けができ助かりました。
調整を頑張ってみます。

最後に、素晴らしいキーボードを開発していただきありがとう御座います。
最高の使用感です。

**Reply by Yowkees** [reply]
Keyballを気に入っていただけて大変光栄です！励みになります。
ぜひ末永くご愛用いただければ幸いです。
